content/blog/2022/2022-02-15-getting-started-with-spring-webflux.md

```diff
@@ -0,0 +1,573 @@
+
+    public WebClient.ResponseSpec getFakeUsers() {
+        return webClient
+                .get()
+                .uri("https://randomuser.me/api/")
+                .retrieve();
+    }
+}
+```
+
+So we called the WebClient instance, made a GET request and retrieved a ResponseSpec instance as a repsonse which can later be converted to Mono or Flux.
+
+Now let’s quickly take a look into one POST call. WebClient allows us to define headers, content types, body, etc. We will just create a simple POST call but we will take a look into functional methods to deal with the backup methods or logic in case of errors or need retries.
+
+```java
+    public Mono<User> postUser(User user) {
+        return webClient
+                .post()
+                .uri("http://localhost:9000/api/users")
+                .header("Authorization", "Basic MY_PASSWORD")
+                .accept(MediaType.APPLICATION_JSON)
+                .body(Mono.just(user), User.class)
+                .retrieve()
+                .bodyToMono(User.class)
+                .log()
+                .retryWhen(Retry.backoff(10, Duration.ofSeconds(2)))
+                .onErrorReturn(new User())
+                .doOnError(throwable -> log.error("Result error out for POST user", throwable))
+                .doFinally(signalType -> log.info("Result Completed for POST User: {}", signalType));
+    }
+```
+
+If we receive the response in the first attempt, then it will simply return the response back. But due to some reason, if the API faces some issue, then it will try to re-attempt for a while after which it can return some indicated data or logs to denote that it has failed due to some reason.
+
+## Alternative Server Used in Webflux
+
+Traditionally, Spring MVC bundles *Tomcat* server for servlet stack applications whereas Spring WebFlux bundles *Reactor Netty* by default for reactive stack applications. *Reactor Netty* is an asynchronous event-driven network application framework built out of Netty server which provides non-blocking and backpressure-ready network engines for HTTP, TCP and UDP clients and servers. Spring WebFlux automatically configures Reactor Netty as the default server but if we want to override the default configuration, we can simply do that by defining them under server prefix.
+
+```yaml
+server:
+  port: 9000
+  http2:
+    enabled: true
+```
+
+We can also define the other properties in the same way that start with the `server` prefix by overriding the default server configuration.
+
+## Concurrency Model
+
+Usually, as we discussed above the traditional Spring Boot applications serve thread-per-request model which means each request to a web server can be handled by different threads allocated per request. However, most of the communications within a given thread are still blocking in nature. Hence, reactive programming came up with a different approach to place a concurrency model that can serve more requests with relatively less number of threads.
```

Comment:
Hmm, reading this section again, I'm not sure what value this section brings over the section "Non-blocking request" earlier in the article. This is pretty much the same thing, is it not? And it doesn't look very specific to Webflux (the description seems to be very generic). If you can make this section provide some specific value, please rewrite it, otherwise, I'd like to remove it.

Reply:
I have removed this section as it looks pretty generic to me as well.